independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java

```diff
@@ -151,26 +164,80 @@ public void handle(Throwable failure) {
         HttpClient httpClient = state.getHttpClient();
         URI uri = state.getUri();
-        boolean isHttps = "https".equals(uri.getScheme());
-        int port = uri.getPort() != -1 ? uri.getPort() : (isHttps ? 443 : 80);
-        RequestOptions requestOptions = new RequestOptions();
-        requestOptions.setHost(uri.getHost());
-        requestOptions.setPort(port);
-        requestOptions.setMethod(HttpMethod.valueOf(state.getHttpMethod()));
-        requestOptions.setURI(uri.getPath() + (uri.getQuery() == null ? "" : "?" + uri.getQuery()));
-        requestOptions.setFollowRedirects(followRedirects);
-        requestOptions.setSsl(isHttps);
         Object readTimeout = state.getConfiguration().getProperty(QuarkusRestClientProperties.READ_TIMEOUT);
-        if (readTimeout instanceof Long) {
-            requestOptions.setTimeout((Long) readTimeout);
+        Uni<RequestOptions> requestOptions;
+        if (uri.getScheme().startsWith("stork")) {
+            boolean isHttps = "storks".equals(uri.getScheme());
+            String serviceName = uri.getHost();
+            Uni<ServiceInstance> serviceInstance;
+            try {
+                serviceInstance = Stork.getInstance().getService(serviceName)
+                        .selectServiceInstance();
+            } catch (Exception e) {
+                return Uni.createFrom().failure(e);
+            }
+            requestOptions = serviceInstance.map(new Function<>() {
+                @Override
+                public RequestOptions apply(ServiceInstance serviceInstance) {
+                    if (serviceInstance.gatherStatistics()) {
+                        state.setCallStatsCollector(serviceInstance);
+                    }
+                    return new RequestOptions()
+                            .setHost(serviceInstance.getHost())
+                            .setPort(serviceInstance.getPort())
+                            .setSsl(isHttps);
+                }
+            });
+        } else {
+            boolean isHttps = "https".equals(uri.getScheme());
+            int port = getPort(isHttps, uri.getPort());
+            requestOptions = Uni.createFrom().item(new RequestOptions().setHost(uri.getHost())
+                    .setPort(port).setSsl(isHttps));
         }
-        return httpClient.request(requestOptions);
+
+        return requestOptions.onItem()
+                .transform(r -> r.setMethod(HttpMethod.valueOf(state.getHttpMethod()))
+                        .setURI(uri.getPath() + (uri.getQuery() == null ? "" : "?" + uri.getQuery()))
+                        .setFollowRedirects(followRedirects))
+                .onItem().invoke(r -> {
+                    if (readTimeout instanceof Long) {
+                        r.setTimeout((Long) readTimeout);
+                    }
+                })
+                .flatMap(options -> AsyncResultUni.toUni(handler -> httpClient.request(options, handler)));
```

Comment:
`flatMap`? I thought one of the points of `Mutiny` was to use proper names and not names that are legacy from functional programming :) Mutiny has better alternatives that this, no?

Reply:
changed to `onItem().transformToUni(...)`

I'm trying to use Mutiny methods everywhere but the well known and established functional method names are much more familiar to me and they sometimes sneak in :)